### PR TITLE
Add nng

### PR DESCRIPTION
**Pull request by jan-janssen**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by carlodri** [comment]
@jan-janssen if you agree I can help as maintainer as a user of this package

**Comment by jan-janssen** [comment]
```
[vcvarsall.bat] Environment initialized for: 'x64'
-- The C compiler identification is MSVC 19.16.27048.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.16.27023/bin/HostX64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Configuring for NNG version 1.5.2
-- Building static libs.
-- Looking for strlcpy
-- Looking for strlcpy - not found
-- Looking for strnlen
-- Looking for strnlen - found
-- Looking for strcasecmp
-- Looking for strcasecmp - not found
-- Looking for strncasecmp
-- Looking for strncasecmp - not found
-- Looking for InitializeConditionVariable
-- Looking for InitializeConditionVariable - found
-- Looking for snprintf
-- Looking for snprintf - found
-- Looking for nl_langinfo
-- Looking for nl_langinfo - not found
-- Test zt disabled (unconfigured)
-- The CXX compiler identification is MSVC 19.16.27048.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.16.27023/bin/HostX64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: C:/bld/nng_1659482876645/work/build
ninja: error: loading 'build.ninja': The system cannot find the file specified.
```

**Comment by jan-janssen** [comment]
I skip the windows tests for now - as I struggle to get `cmake` and `ninja` to collaborate on windows. Still any recommendations are welcome.

**Comment by jan-janssen** [comment]
@conda-forge/staged-recipes, ready for review

**Comment by jan-janssen** [comment]
Windows is now also working

File: recipes/nng/build.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+else
+	cmake -B build -S . \
+		${CMAKE_ARGS} \
+                -G Ninja \
+		-DCMAKE_INSTALL_PREFIX=$PREFIX \
+		-DCMAKE_INSTALL_LIBDIR=lib \
+		-DCMAKE_BUILD_TYPE=Release \
+		-DCMAKE_VERBOSE_MAKEFILE=ON
+fi
+
```
Comment (ocefpaf) [inline recipes/nng/build.sh:23]: Is there a ctest/make check or something that we can run here?

Comment (jan-janssen) [reply recipes/nng/build.sh:23]: The issue is there is one test failing: 
```
      Start 11: nng.platform.resolver_test
11/91 Test #11: nng.platform.resolver_test ..................***Failed    0.03 sec
```

**Comment by carlodri** [comment]
Thanks @ocefpaf and @jan-janssen!

**Pull request merged**